src/_actions/category.actions.js

```diff
@@ -28,6 +31,32 @@ export const loadCategories = () => {
  *
  * @return {object} type and payload
  */
-export const loadCategoriesSuccess = (categories) => {
-  return { type: LOAD_CATEGORIES_SUCCESS, payload: categories };
-}
+export const loadCategoriesSuccess = categories =>
+  ({ type: LOAD_CATEGORIES_SUCCESS, payload: categories });
+
+export const createCategory = newCategory => dispatch =>
+  axios.post('categories', newCategory).then((response) => {
+    dispatch(createCategorySuccess(response.data));
+    dispatch(updateToastMessageContent('New Category Saved Successfully',
+      'success'));
+  }).catch((error) => {
+    dispatch(createCategoryFailure(error));
+    dispatch(updateToastMessageContent('Could Not Save The Category',
+      'error'));
+  });
+
+export const createCategorySuccess = category => ({
+  type: CREATE_CATEGORY_SUCCESS,
+  payload: category });
+export const createCategoryFailure = error => ({
+  type: CREATE_CATEGORY_FAILURE,
+  payload: error });
+export const updateToastMessageContent = (message, type) => (
+  {
```

Comment:
Nitpick: A little bit of restructuring needed for aesthetics 🙂 . Move the opening brace to previous line. The action creator should be structured as shown in the snippet

```
export const updateToastMessageContent = (message, type) => ({
  type: UPDATE_TOAST_MESSAGE_CONTENT,
  payload: {
    message,
    type
  }
});

Reply:
Done.